server/controllers/finance/reports/income_expense/index.js

```diff
@@ -77,6 +77,10 @@ function document(req, res, next) {
 
       const root = tree.getRootNode();
 
+      if (!Array.isArray(root.children) || root.children.length < 2) {
+        throw new BadRequest('Could not find both income and expense accounts for the time period');
+      }
```

Comment:
Could you add a key as second parameter of BadRequest() for having a meaningful message for the user.

Ex.
```js
throw new BadRequest('Could not find both income and expense accounts for the time period', 'ERRORS.NO_TRANSACTION_FOUND');
```

Reply:
Or 'ERRORS.EVERYTHING_BAD' 😄 

Will do!